### DISCUSSION TITLE
Does anyone know why only SOME HTML tags work on your README?

### DISCUSSION THREAD

**Discussion by nikoniko78** [community/community#144854] [Programming Help] [answered]
### Body

Certain tags wont work, for example, the float: tag does not work however the text-align: tag does. Any fixes?

### Guidelines

- [X] I have read and understood this category's [guidelines](https://github.com/orgs/community/discussions/51384) before making this post.

**Comment by peterbikes** [comment, marked as answer]
GitHub READMEs are rendered using GitHub Flavored Markdown (GFM), which supports a limited subset of HTML tags and CSS properties for security and simplicity.
However, basic inline styles like text-align often work because they're less intrusive. Unfortunately, there's no way to enable unsupported tags or styles in a GitHub README directly.
Use Markdown formatting wherever possible.
If you need complex layouts or styles, consider hosting your documentation elsewhere (e.g., GitHub Pages) where you have full control over HTML and CSS.

Here are some supported HTML Tags in GitHub READMEs:

- Headings:

`<h1>, <h2>, <h3>, <h4>, <h5>, <h6>`

- Text Formatting:

`<b>, <strong>, <i>, <em>,<u>, <s>, <strike>, <del> , <br> , <sub>, <sup>, <code>, <pre>`

- Links and Media:

`<a>, <img>`

- Lists:

` <ul>, <ol>, <li>`

- Tables:

` <table>, <thead>, <tbody>, <tr>, <td>, <th>`

- Others:

`<blockquote> (block quotes)` and `<hr> (horizontal rule)`


HTML elements like ` <style>, <script>, <iframe>, and <div> ` are not supported for security reasons or because they can disrupt layout consistency. Some basic inline styles (like color or text-align) may work, but most advanced CSS (like float) won't be applied.

**Reply by NSExceptional** [reply]
Where can I find the docs on this?

**Comment by nikoniko78** [comment]
Thanks! I noticed this while trying to snap a badge to the right side of the README while also keeping it on the same line as some text. Do you have any idea on how to achieve this?

**Comment by peterbikes** [comment]
That's a good question, i am not sure if it would work or not, but I would try something like this:

`<p>Text on the left <img align="right" src="link/to/the/picture"></p>`

**Reply by abdelillahCoder** [reply]
Your code will work partially, but the `align` attribute for images is outdated and no longer supported in modern HTML standards. Instead, you should use CSS to achieve the desired layout. Here's how you can update it:

html
<p>
  Text on the left 
  <img style="float: right; margin-left: 10px;" src="link/to/the/picture" alt="Image description">
</p>


 Explanation:
-`float: right;: Makes the image float to the right.
- margin-left: 10px;: Adds some space between the text and the image.
- `alt` attribute Provides a description of the image for accessibility.

**Comment by Gesronet** [comment]
The reason only some HTML tags work in your README is that platforms like GitHub render README files using a specific markup language (like Markdown) that only supports a subset of HTML. This is primarily for security and consistency reasons.
